### PR TITLE
internal: Remove Enzyme from visx

### PR DESCRIPTION
**Pull request by chazcb**
This PR replaces Enzyme with RTL in prep for React 19 support.

#### :house: Internal

- Replaces Enzyme with RTL while avoiding `jest.mock` usage

### CODE REVIEW

File: packages/visx-legend/test/scales.test.tsx [hunk 1]
```diff
@@ -18,9 +19,14 @@ describe('Legend scales', () => {
       range: [1, 5, 10, 15, 20],
     });
 
-    expect(() => shallow(<LegendLinear scale={linearScale} />)).not.toThrow();
-    expect(() => shallow(<LegendSize scale={linearScale} />)).not.toThrow();
-    expect(() => shallow(<Legend scale={linearScale} />)).not.toThrow();
+    const { container: container1 } = render(<LegendLinear scale={linearScale} />);
+    expect(container1).toBeInTheDocument();
```
Comment (williaster) [inline packages/visx-legend/test/scales.test.tsx:23]: probably okay, could use `.not.toThrow()`

File: packages/visx-marker/test/Arrow.test.tsx [hunk 2]
```diff
@@ -1,38 +1,56 @@
...
+      <svg>
+        <defs>
+          <MarkerArrow id="marker-circle-test" />
+        </defs>
+      </svg>,
+    );
+
+    const marker = container.querySelector('marker');
+    expect(marker).toBeInTheDocument();
+    expect(marker).toHaveAttribute('id', 'marker-circle-test');
```
Comment (williaster) [inline packages/visx-marker/test/Arrow.test.tsx:23]: lost some of the test here, but now in the next test. just change name?

File: packages/visx-marker/test/Line.test.tsx [hunk 3]
```diff
@@ -1,37 +1,54 @@
...
-    expect(marker).toHaveLength(1);
-    expect(rect).toHaveLength(1);
+  test('it should render a marker element', () => {
+    const { container } = render(
+      <svg>
+        <MarkerLine id="marker-line-test" />
+      </svg>,
+    );
+
+    const marker = container.querySelector('marker');
```
Comment (williaster) [inline packages/visx-marker/test/Line.test.tsx:19]: lost the rect test, but now in next test

File: packages/visx-pattern/test/PatternLines.test.tsx [hunk 4]
```diff
@@ -1,50 +1,75 @@
...
+    const { container } = render(
+      <SVGWrapper>
+        <PatternLines id="test" height={4} width={4} background="blue" />
+      </SVGWrapper>,
+    );
+
+    const pattern = container.querySelector('pattern');
+    expect(pattern).toBeInTheDocument();
+
+    const backgroundRect = container.querySelector('.visx-pattern-line-background');
```
Comment (williaster) [inline packages/visx-pattern/test/PatternLines.test.tsx:37]: could be a rect

File: packages/visx-react-spring/test/AnimatedTicks.test.tsx [hunk 5]
```diff
@@ -44,8 +47,15 @@ describe('AnimatedTicks', () => {
...
             { from: { x: 0, y: 0 }, to: { x: 0, y: 5 }, value: 0, index: 0, formattedValue: '0' },
           ]}
-        />,
-      ),
-    ).not.toThrow();
+        />
+      </svg>,
+    );
+
+    const svgElement = container.querySelector('svg');
```
Comment (williaster) [inline packages/visx-react-spring/test/AnimatedTicks.test.tsx:54]: unnecessary

File: packages/visx-react-spring/test/AnimatedGridRows.test.tsx [hunk 6]
```diff
@@ -1,17 +1,58 @@
...
-      ),
-    ).not.toThrow();
+
+  it('should render without crashing', () => {
+    const { container } = render(
+      <svg>
+        <AnimatedGridRows {...defaultProps} />
+      </svg>,
+    );
+    expect(container.querySelector('svg')).toBeInTheDocument();
```
Comment (williaster) [inline packages/visx-react-spring/test/AnimatedGridRows.test.tsx:28]: unnecessary

File: packages/visx-shape/test/Area.test.tsx [hunk 7]
```diff
@@ -1,10 +1,8 @@
 import React from 'react';
-import { shallow } from 'enzyme';
 import { render } from '@testing-library/react';
 import '@testing-library/jest-dom';
-
+import { Area as D3Area } from 'd3-shape';
```
Comment (williaster) [inline packages/visx-shape/test/Area.test.tsx:4]: hmm, ideally don't need this

File: packages/visx-shape/test/AreaClosed.test.tsx [hunk 8]
```diff
@@ -1,11 +1,9 @@
 import React from 'react';
-import { shallow } from 'enzyme';
 import { render } from '@testing-library/react';
 import '@testing-library/jest-dom';
-
 import { scaleLinear } from '@visx/scale';
+import { Area } from 'd3-shape';
```
Comment (williaster) [inline packages/visx-shape/test/AreaClosed.test.tsx:5]: avoid?

File: packages/visx-shape/test/BarGroup.test.tsx [hunk 9]
```diff
@@ -28,92 +20,66 @@ const data: Datum[] = [
...
+    expect(container.querySelector('.visx-bar-group')).toBeInTheDocument();
   });
 
   test('it should set className prop', () => {
-    const wrapper = BarGroupWrapper({ className: 'test' });
-    expect(wrapper.prop('className')).toBe('visx-bar-group test');
+    const { container } = renderWithSvg(<BarGroup {...defaultProps} className="test" />);
+    const element = container.querySelector('g.visx-bar-group');
+    expect(element).toHaveClass('test');
+    expect(element).toHaveClass('visx-bar-group');
```
Comment (williaster) [inline packages/visx-shape/test/BarGroup.test.tsx:50]: unnecessary

File: packages/visx-shape/test/BarStackHorizontal.test.tsx [hunk 10]
```diff
@@ -17,53 +17,67 @@ describe('<BarStackHorizontal />', () => {
...
+          color={(d) => d}
+          keys={[]}
+        />
+      </svg>,
     );
-    expect(wrapper.prop('className')).toBe('visx-bar-stack-horizontal');
+
+    const element = container.querySelector('.visx-bar-stack-horizontal');
+    expect(element).toBeInTheDocument();
+    expect(element).toHaveClass('visx-bar-stack-horizontal');
```
Comment (williaster) [inline packages/visx-shape/test/BarStackHorizontal.test.tsx:37]: unnecessary

File: packages/visx-shape/test/Pie.test.tsx [hunk 11]
```diff
@@ -44,90 +44,112 @@ const browserUsage: Datum[] = [
...
   test('it should not break on sort callbacks', () => {
-    expect(() => {
-      PieWrapper({ pieSort: () => 0, pieSortValues: () => 0 });
-    }).not.toThrow();
+    const { container } = render(
+      <svg>
+        <Pie data={browserUsage} pieSort={() => 0} pieSortValues={() => 0} />
+      </svg>,
+    );
+    expect(container).toBeInTheDocument();
```
Comment (williaster) [inline packages/visx-shape/test/Pie.test.tsx:58]: expect `.not.toThrow()`?

File: packages/visx-xychart/test/components/AnnotationLabel.test.tsx [hunk 12]
```diff
@@ -1,14 +1,71 @@
...
+    // then foreignObject must exist and be working correctly
+    const title = getByText('Title in foreignObject');
+    const subtitle = getByText('Subtitle in foreignObject');
+
+    expect(title).toBeInTheDocument();
+    expect(subtitle).toBeInTheDocument();
+
+    // Verify the text is within an SVG element
+    expect(title.closest('svg')).toBeInTheDocument();
+    expect(subtitle.closest('svg')).toBeInTheDocument();
```
Comment (williaster) [inline packages/visx-xychart/test/components/AnnotationLabel.test.tsx:69]: maybe unnecessary

File: packages/visx-xychart/test/components/AnnotationConnector.test.tsx [hunk 13]
```diff
@@ -1,14 +1,39 @@
...
+
+    // Verify the path has correct attributes
+    expect(path).toHaveAttribute('d', expect.any(String));
+    expect(path).toHaveAttribute('stroke', expect.any(String));
+    expect(path).toHaveAttribute('fill', 'transparent');
+
+    // Verify the path connects the correct points
+    const dAttribute = path?.getAttribute('d') || '';
+    expect(dAttribute).toContain('M100,100'); // Should start at x,y
+    expect(dAttribute).toContain('150,150'); // Should end at x+dx,y+dy
```
Comment (williaster) [inline packages/visx-xychart/test/components/AnnotationConnector.test.tsx:37]: improved the tests here! 🤖 🎉

File: packages/visx-tooltip/test/useTooltipInPortal.test.tsx [hunk 14]
```diff
@@ -14,30 +15,49 @@ const TooltipWithZIndex = ({ zIndexOption, zIndexProp }: TooltipWithZIndexProps)
...
+  it('should pass zIndex prop from component to Portal', async () => {
+    const { baseElement } = render(
       <TooltipWithZIndex zIndexOption={1} zIndexProp="var(--tooltip-zindex)" />,
+    );
+
+    await waitFor(
+      () => {
+        const portalDiv = baseElement.querySelector('[style*="z-index: var(--tooltip-zindex)"]');
+        expect(portalDiv).toBeInTheDocument();
+        expect(portalDiv).toHaveStyle('z-index: var(--tooltip-zindex)');
```
Comment (williaster) [inline packages/visx-tooltip/test/useTooltipInPortal.test.tsx:55]: unnecessary?

File: packages/visx-tooltip/test/Tooltip.test.tsx [hunk 15]
```diff
@@ -8,18 +8,37 @@ describe('<Tooltip />', () => {
...
-    const wrapper = shallow(<Tooltip>Hello</Tooltip>);
-    const styles = wrapper.props().style;
+    const { container } = render(<Tooltip>Hello</Tooltip>);
+    const tooltip = container.firstChild as HTMLElement;
+    const computedStyle = window.getComputedStyle(tooltip);
+
     Object.entries(defaultStyles).forEach(([key, value]) => {
-      expect(styles[key]).toBe(value);
+      // For colors, compare the computed style which will be normalized
+      if (key === 'backgroundColor' || key === 'color') {
```
Comment (williaster) [inline packages/visx-tooltip/test/Tooltip.test.tsx:17]: could this be simplified to any string?

**Review by williaster — COMMENTED** [review]
Thanks @chazcb , this is awesome 👏 🤖 

Noticed some minor problems I can fix.

File: packages/visx-text/package.json [hunk 16]
```diff
@@ -40,8 +40,5 @@
   },
   "publishConfig": {
     "access": "public"
-  },
-  "devDependencies": {
-    "@testing-library/react-hooks": "^8.0.0"
```
Comment (williaster) [inline packages/visx-text/package.json:45]: left over from https://github.com/airbnb/visx/pull/1890, hoping this fixes the failing `@visx/react-spring` hooks test suite

**Review by williaster — APPROVED** [review]
we did it finish 🏁

**Comment by darthmaim** [comment]
This is really amazing 🎉

**Pull request merged**